pkg/master/client_ca_hook.go

```diff
@@ -38,25 +41,38 @@ type ClientCARegistrationHook struct {
 }
 
 func (h ClientCARegistrationHook) PostStartHook(hookContext genericapiserver.PostStartHookContext) error {
+	// no work to do
 	if len(h.ClientCA) == 0 && len(h.RequestHeaderCA) == 0 {
 		return nil
 	}
 
-	client, err := coreclient.NewForConfig(hookContext.LoopbackClientConfig)
+	// intializing CAs is important so that aggregated API servers can come up with "normal" config.
+	// We've seen lagging etcd before, so we want to retry this a few times before we decide to crashloop
+	// the API server on it.
+	err := wait.Poll(1*time.Second, 30*time.Second, func() (done bool, err error) {
+		client, err := coreclient.NewForConfig(hookContext.LoopbackClientConfig)
```

Comment:
> why retry the NewForConfig? If it important, comment please.

If memory serves, a new client sometimes makes a discovery request.  I remember hitting in the RBAC post start hook.